### PR TITLE
[SKIP] Update Benchmark Results (Azure F4s-v2-4vCPU-8GB-RAM)

### PR DESCRIPTION
**Pull request by jinyus**
Automated PR

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -64,53 +64,54 @@ docker run -e TEST_NAME=all -it --rm databench
...
+| _Julia HO_[^1] | 8.15 ms | 54.00 ms | 138.00 ms | 200.15 ms |
+| D | 24.73 ms | 315.66 ms | 2.70 s | 3.04 s |
+| Nim | 22.15 ms | 317.63 ms | 2.84 s | 3.18 s |
+| Go | 22.69 ms | 322.86 ms | 2.86 s | 3.21 s |
+| Zig | 23.00 ms | 337.00 ms | 2.99 s | 3.35 s |
+| Rust | 23.53 ms | 340.58 ms | 3.03 s | 3.40 s |
+| C# (AOT) | 26.34 ms | 373.28 ms | 3.26 s | 3.65 s |
+| C# (JIT) | 27.52 ms | 376.84 ms | 3.26 s | 3.67 s |
+| F# (AOT) | 27.38 ms | 392.00 ms | 3.42 s | 3.84 s |
+| F# (JIT) | 91.46 ms | 532.33 ms | 4.04 s | 4.67 s |
```
Comment (Moelf) [inline readme.md:76]: What happened to the first time here... Is vCPU really this unstable

**Comment by Moelf** [comment]
As I said before we should use minimal time as a proxy to performance because VM is too unstable

**Comment by Moelf** [comment]
that metric on a proper machine should still be closer to the minimal time than to this vCPU "average" time.

**Comment by Moelf** [comment]
I mean compare the Julia time to just a few hours ago:
https://github.com/jinyus/related_post_gen/pull/400#issuecomment-1789873269

**Comment by algunion** [comment]
> I mean compare the Julia time to just a few hours ago on the exact same code base and machine: [#400 (comment)](https://github.com/jinyus/related_post_gen/pull/400#issuecomment-1789873269)
> 
> stable 24ms to this PR 31ms

Comparing what is happening now with the original `StaticArrays` implementation (and given there is the same VM involved), I think the culprit is related to Julia/implementation. 

When comparing the two - I can see a much higher variability in the recent implementations (e.g., before we only had ~1ms).

P. S. We can easily put this to the test by actually reviving that (exact) PR and stop trying to do all kinds of esthetic changes (that we assume will not impact the perf).

**Comment by cyrusmsk** [comment]
I understand that Julia has some problems with that but consider those 2 cases:
Case 1:
Results of the processing
25ms
24ms
25ms
26ms
16ms
24ms
25ms
26ms

Who knows what happened there… cache, cpu registers, operating system, file system, miracle..
But showing 16ms as a result is a misleading in my opinion.

Case 2:
Results of the processing:
25ms
24ms
26ms
25ms
1000ms
25ms
24ms

should this outlier be calculated? Like one client will wait his response too long. We definitely should be aware of those behaviour, but probably not count them in benchmark result.

that’s why maybe mean is not the best metric and we could use median or mode instead

**Comment by Moelf** [comment]
>Who knows what happened there… cache, cpu registers, operating system, file system, miracle..
But showing 16ms as a result is a misleading in my opinion.

whatever crap happened can only make time slower :)

**Comment by Tortar** [comment]
This is not noise in this case @Moelf, I think you agree that something is strange about Julia, and as such I don't understand why we should favor it using the minimum. In any case the VM is showing the same behaviour of my two computers attested in https://github.com/JuliaLang/julia/issues/51988, these are the results of the last run in Julia:

```
Benchmark 1: julia --startup-file=no --project=Related -e "using Related; main()"
Processing time (w/o IO): 32 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 32 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 30 milliseconds
Processing time (w/o IO): 30 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
Processing time (w/o IO): 31 milliseconds
  Time (mean ± σ):      1.020 s ±  0.012 s    [User: 0.893 s, System: 0.231 s]
  Range (min … max):    0.998 s …  1.035 s    10 runs
```

so using the minimum will not help in this case either. But it shouldn't be done anyway since I don't think that oscillations are just noise since if you look at all other languages they don't seem to oscillate as much as Julia.

**Comment by Tortar** [comment]
one thing to try is to see if these oscillations are only related to StaticArrays

**Comment by Tortar** [comment]
The answer is no...Incredibly it happens less often that precompilation goes bad with `Vector`, but it happens!

File: readme.md (hunk 1 above)
Comment (jinyus) [reply readme.md:76]: Are you referencing f#? It has consistently gotten that score over the last ~10 runs so I consider that stable. The jit just performs better on longer runs.

**Comment by jinyus** [comment]
Doing fresh run right now

**Pull request closed without merging**